src/mock.js

```diff
@@ -187,6 +200,7 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
     }
     // eslint-disable-next-line no-param-reassign
     field.resolve = mockType(field.type, typeName, fieldName);
+    assignResolveType(field.type);
```

Comment:
One more thing: can you wrap this in an if statement so it only runs if `preserveResolvers` is false?